render/egl.c

```diff
@@ -413,6 +416,40 @@ EGLImageKHR wlr_egl_create_image_from_dmabuf(struct wlr_egl *egl,
 		EGL_LINUX_DMA_BUF_EXT, NULL, attribs);
 }
 
+bool wlr_egl_export_image_to_dmabuf(struct wlr_egl *egl, EGLImageKHR image,
+		int32_t width, int32_t height, uint32_t flags,
+		struct wlr_dmabuf_buffer_attribs *attribs) {
+	memset(attribs, 0, sizeof(struct wlr_dmabuf_buffer_attribs));
+
+	if (!egl->egl_exts.dmabuf_export || !eglExportDMABUFImageQueryMESA ||
+			!eglExportDMABUFImageMESA) {
+		return false;
+	}
+
+	EGLuint64KHR modifier;
```

Comment:
Should this be 
```
EGLuint64KHR modifiers[WLR_LINUX_DMABUF_MAXPLANES}; 
```
since the modifiers are per plane?

Reply:
The first call assumes the caller doesn't know how many planes the image has, so there's only one modifier. This is a limitation of the EGL API I guess.

Reply:
Looking at mesa it's an array:

https://sources.debian.org/src/mesa/17.3.9-1/src/gallium/drivers/etnaviv/etnaviv_screen.c/?hl=591#L604

Reply:
The EGL docs say that when making the `eglExportDMABUFImageQueryMESA` call the caller doesn't know about the number of planes. The `eglExportDMABUFImageMESA` call is used to retrieve per-plane properties.

Reply:
Sorry for being unclear, the main thing I'm trying to say is that the loop in https://github.com/swaywm/wlroots/pull/906/files#diff-7d28873cf02c85696a511bcb3a1521daR439 should use per plane modifiers.

Reply:
Hmm, I still don't get it x)

Are per-plane modifiers encoded in this single int somehow?

Reply:
Note the plural form ```modifiers``` in https://www.khronos.org/registry/EGL/extensions/MESA/EGL_MESA_image_dma_buf_export.txt and how https://sources.debian.org/src/mesa/17.3.9-1/src/gallium/drivers/etnaviv/etnaviv_screen.c/?hl=591#L614 fills in the values per plane.

Reply:
I think this function is used for `eglQueryDmaBufModifiersEXT`, not for `eglExportDMABUFImageQueryMESA`.

Reply:
See below, I mixed up things. Sorry for the confusion.

Reply:
No worries, this was confusing for me too!